libraries/app/database_api.cpp

```diff
@@ -362,6 +363,24 @@ optional<block_header> database_api_impl::get_block_header(uint32_t block_num) c
       return *result;
    return {};
 }
+map<uint32_t,block_header> database_api::get_block_header_batch(const vector<uint32_t> block_num)const
+{
+   return my->get_block_header_batch( block_num );
+}
+
+map<uint32_t,block_header> database_api_impl::get_block_header_batch(const vector<uint32_t> block_num) const
+{
+   map<uint32_t,block_header> results;
+   for (const uint32_t& i : block_num) {
+      auto result = _db.fetch_block_by_number(i);
+      if (result)
+         results[i] = *result;
```

Comment:
thank you for the comment @xeroc . you might be right but i used the same code as get_block_header:

```
optional<block_header> database_api_impl::get_block_header(uint32_t block_num) const
{
   auto result = _db.fetch_block_by_number(block_num);
   if(result)
      return *result;
   return {};
}
```
adding @vikramrajkumar can probably help decide.